src/rust/engine/process_executor/src/main.rs

```diff
@@ -205,86 +156,22 @@ struct Opt {
 ///
 /// It does not perform $PATH lookup or shell expansion.
 #[tokio::main]
-async fn main() {
+async fn main() -> Result<(), String> {
```

Comment:
Easier to return a string error from main(), which rust will show as an error on the CLI, rather than calling panic all over the place.